Library/Homebrew/formula.rb

```diff
@@ -261,6 +261,15 @@ def installed?
     (dir = installed_prefix).directory? && dir.children.length > 0
   end
 
+  # If this {Formula} has been modified by the user.  Relies on git.
+  def adulterated?
+    repo = core_formula? ? HOMEBREW_LIBRARY : path_of_repository
+    repo.cd do
```

Comment:
I originally just targeted things in `HOMEBREW_LIBRARY` and added `path_of_repository` later, and didn't refactor down.